### PR TITLE
Replaces VopVector with VnodeBackend

### PR DESCRIPTION
**Pull request by ultimaweapon**
So it is much more idiomatic to Rust. The benefits we got from this is we no longer need to cast `v_data` to the FS specific data.

### CODE REVIEW

**Review by SuchAFuriousDeath — APPROVED** [review]

**Comment by SuchAFuriousDeath** [comment]
Hmmm. won't we need to call the default methods separately from other parts of the code? They should probably be standalone methods and the default implementation in the trait should call those methods.

**Comment by SuchAFuriousDeath** [comment]
If you merge main into this, I'll merge it and deal with conflicts in my branches.

**Comment by ultimaweapon** [comment]
> Hmmm. won't we need to call the default methods separately from other parts of the code? They should probably be standalone methods and the default implementation in the trait should call those methods.

What are the benefits of that?

**Comment by ultimaweapon** [comment]
Sorry I missed the first part. It is unlikely that we need to call the default implementation outside. If that happens we can extract it later. For now it make the code hard to follow without any benefits.

**Comment by SuchAFuriousDeath** [comment]
Yeah I think you're right. I confused the default ops with the handlers in vnodeop_desc
